### PR TITLE
feat: add custom header for mm downloader.

### PR DESCRIPTION
**Pull request by JJMeg**
## Description

Allow callers to attach custom HTTP headers (e.g., Authorization, Referer) when downloading multimodal resources via image_url/video_url/audio_url. Headers can be set at two levels with clear precedence:

  1. Per-request headers via the API (highest priority)
  2. Service-level defaults via --mm_download_headers gflag (fallback)
  
When both are present, per-request headers override the global defaults on a per-key basis; non-overlapping keys from both levels are merged.

**1. Service-level gflag**
`--mm_download_headers (string, default: "")`  Global HTTP headers applied to all multimodal downloads. Accepts a JSON  object string. Parsed once at first use and cached for the process lifetime.

Example:
```shell
    --mm_download_headers={"Authorization":"Bearer global","Referer":"https://example.com"}
```

**2. Per-request API format**
```json
{
	"messages": [{
		"role": "user",
		"content": [{
			"type": "image_url",
			"image_url": {
				"url": "http://img.example.com/photo.jpg",
				"headers": {
					"Authorization": "token123"
				}
			}
		}]
	}]
}
```

**3. Merge behavior**
```text
--mm_download_headers={"Referer":"https://default.com"}
  +
  Request: {"headers": {"Authorization": "Bearer xxx"}}
  =
  Effective:  Authorization=Bearer xxx, Referer=https://default.com
```
### Changes:
proto:
    - multimodal.proto: add map<string, string> headers to ImageURL, VideoURL, and AudioURL messages

core:
    - message.h: add std::unordered_map<std::string, std::string> headers to ImageURL, VideoURL, and AudioURL structs
    - global_flags.h/cpp: declare and define --mm_download_headers gflag
    - http_downloader.h/cpp:
        - parse_headers_json() — parse JSON header string into map
        - parse_global_headers() — cached parser for the gflag
        - BRpcDownloader::download() — apply global defaults first, then per-request headers via cntl.http_request().SetHeader()
    - mm_handler.h/cpp: thread headers from MMContent through load_from_http() in ImageHandler, VideoHandler, and AudioHandler
    - mm_service_utils.h: parse headers from proto into C++ structs in build_messages()

tests:
    - http_downloader_test.cpp: unit tests covering parse_headers_json()
      (empty/valid/invalid/edge-case JSON), struct default & assignment,
      MMContent header plumbing, and header merge priority order

## Change Type

- [ ] Bug fix
- [x] New feature
- [ ] Performance improvement
- [ ] Refactor
- [ ] Documentation
- [ ] Test
- [ ] Build or CI

## Pull Request Checklist

Thank you for contributing to xLLM. Before requesting review, please make sure the following items are complete.

### PR Title and Commit Messages

- [x] The PR title and each commit message follow the xLLM commit format: `<type>: <subject>`.

> Allowed types: feat, bugfix, docs, test, refactor, chore, style, revert, perf, model, build, release.
> The subject should use clear English, start with a verb, include at least 4 words, and end with `.`.

### Pre-commit Checks

- [x] I have installed `pre-commit` by running `pip install pre-commit` or an equivalent command.
- [x] I have installed the hooks with `pre-commit install`.
- [x] I have run `pre-commit run --all-files` and fixed any reported issues.

> If you are unsure how to set up `pre-commit`, see [the pre-commit documentation](https://pre-commit.com/).

### Self Review

- [x] I have self-reviewed the code according to `.agents/skills/code-review/references/custom-code-style.md`, especially code written or assisted by AI.
- [x] I have rebased this PR onto the latest `main` branch.

### Build and Test Coverage

- [ ] Tests have been added or updated as needed.
- [x] CUDA: `python setup.py build test` has passed on a CUDA machine.
- [x] NPU: `python setup.py build test` has passed on an NPU machine.
- [x] MLU: `python setup.py build test` has passed on an MLU machine.

### CODE REVIEW

File: xllm/core/util/http_downloader.cpp [hunk 1]
```diff
@@ -77,11 +80,16 @@ std::shared_ptr<brpc::Channel> BRpcDownloader::get_channel(
...
-                              const std::string& url,
-                              std::string& data) {
+bool BRpcDownloader::download(
+    const std::string& host,
+    const std::string& url,
+    std::string& data,
+    const std::unordered_map<std::string, std::string>& headers) {
   brpc::Controller cntl;
   cntl.http_request().uri() = url;
+  for (const auto& [k, v] : headers) {
```
Comment (xiao-yu-chen) [inline xllm/core/util/http_downloader.cpp:121]: does the gflags at the service level meet the requirements？

Comment (JJMeg) [reply xllm/core/util/http_downloader.cpp:121]: No, we don't rely on gflags for this. Instead, headers are passed at the per-request level through the API, which is more flexible than a service-level flag.

File: xllm/core/util/http_downloader.cpp [hunk 2]
```diff
@@ -15,17 +15,59 @@ limitations under the License.
 
 #include "http_downloader.h"
 
+#include <gflags/gflags.h>
 #include <glog/logging.h>
 
+#include <nlohmann/json.hpp>
+
+DEFINE_string(mm_download_headers,
```
Comment (xiao-yu-chen) [inline xllm/core/util/http_downloader.cpp]: It needs to be defined in the config module，and use config instead of gflags。

File: xllm/core/framework/config/model_config.cpp [hunk 3]
```diff
@@ -75,6 +75,12 @@ DEFINE_bool(enable_return_mm_full_embeddings,
             false,
             "return vit and sequence embeddings for vlm models");
 
+DEFINE_string(mm_download_headers,
```
Comment (XuZhang99) [inline xllm/core/framework/config/model_config.cpp:78]: not declare in `global_flags.h`?

Comment (JJMeg) [reply xllm/core/framework/config/model_config.cpp:78]: Fixed

**Review by xiao-yu-chen — DISMISSED** [review]

**Review by XuZhang99 — DISMISSED** [review]

**Review by XuZhang99 — APPROVED** [review]

**Review by xiao-yu-chen — APPROVED** [review]
